File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    int OVENT_TIME = 40;
    int TIME_PER_LAYER = 2;
    
    public int expectedMinutesInOven() {
        return OVENT_TIME;
    }
    
    public int remainingMinutesInOven(int actualTimeInOven) {
        return expectedMinutesInOven() - actualTimeInOven;
    }

    public int preparationTimeInMinutes(int numberOfLayers) {
        return TIME_PER_LAYER * numberOfLayers;
    }

    public int totalTimeInMinutes(int numberOfLayers, int actualTimeInOven) {
        return preparationTimeInMinutes(numberOfLayers) + actualTimeInOven;
    }
}

File: solutions/lua/lasagna/1/lasagna.lua
local lasagna = {}

lasagna.oven_time = 40
lasagna.time_per_layer = 2

-- returns the remaining minutes based on the actual minutes in the oven
function lasagna.remaining_oven_time(actual_minutes_in_oven)
  return lasagna.oven_time - actual_minutes_in_oven
end

-- calculates and returns the time needed to prepare the lasagna on the amount
-- of layers
function lasagna.preparation_time(number_of_layers)
  return number_of_layers * lasagna.time_per_layer
end

-- calculates the time elapsed cooking the lasagna - including preparation time
-- and baking time.
function lasagna.elapsed_time(number_of_layers, actual_minutes_in_oven)
  return  lasagna.preparation_time(number_of_layers) + actual_minutes_in_oven
end

return lasagna
